### PR TITLE
msprime package

### PR DESCRIPTION
**Pull request by ilbiondo**
New package for msprime

### CODE REVIEW

File: var/spack/repos/builtin/packages/msprime/package.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Msprime(PythonPackage):
+    """The library is a reimplementation of Hudson’s seminal ms program,
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/msprime/package.py]:
```suggestion
    """The library is a reimplementation of Hudson's seminal ms program,
```
This should fix failing Python 2 tests

File: var/spack/repos/builtin/packages/msprime/package.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Msprime(PythonPackage):
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/msprime/package.py:9]: Any reason not to name this `py-msprime` like all other Python packages?

File: var/spack/repos/builtin/packages/msprime/package.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    and aims to eventually reproduce all its functionality."""
+
+    homepage = "https://msprime.readthedocs.io/en/stable/"
+    url      = "https://pypi.io/packages/source/m/msprime/msprime-0.7.4.tar.gz"
+
+    version('0.7.4',   sha256='cbbee83879444d99c0a79b9cc3688ff74c7c39426d92134d0ccf17a1e15716ba')
+
+    depends_on('python@3:',     type=('build', 'run'))
+    depends_on('py-setuptools', type=('build', 'run'))
+    depends_on('gsl',           type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/msprime/package.py]:
```suggestion
    depends_on('gsl',           type=('build', 'link'))
```

File: var/spack/repos/builtin/packages/msprime/package.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    depends_on('tskit',         type='run')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/msprime/package.py]:
```suggestion
    depends_on('py-tskit',      type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/msprime/package.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    depends_on('py-svgwrite',   type='run')
+    depends_on('py-jsonschema', type='run')
+    depends_on('py-h5py',       type='run')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/msprime/package.py]: I don't see these dependencies listed anywhere. I do see a missing dependency on numpy 1.7+

File: var/spack/repos/builtin/packages/msprime/package.py [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+class Msprime(PythonPackage):
+    """The library is a reimplementation of Hudson’s seminal ms program,
+    and aims to eventually reproduce all its functionality."""
+
+    homepage = "https://msprime.readthedocs.io/en/stable/"
+    url      = "https://pypi.io/packages/source/m/msprime/msprime-0.7.4.tar.gz"
+
+    version('0.7.4',   sha256='cbbee83879444d99c0a79b9cc3688ff74c7c39426d92134d0ccf17a1e15716ba')
+
+    depends_on('python@3:',     type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/msprime/package.py]:
```suggestion
    depends_on('python@3.4:',   type=('build', 'run'))
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/msprime/package.py (hunk 2 above)
Comment (ilbiondo) [reply var/spack/repos/builtin/packages/msprime/package.py:9]: As for tskit, it's a standalone tool so the py- isn't adapted.

https://spack.readthedocs.io/en/latest/build_systems/pythonpackage.html#choosing-a-package-name

File: var/spack/repos/builtin/packages/msprime/package.py (hunk 6 above)
Comment (ilbiondo) [reply var/spack/repos/builtin/packages/msprime/package.py]: 3.6 according to the setup.py - fixed in 637c32b

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/msprime/package.py]: That's in master/develop, not in 0.7.4.

**Comment by alalazo** [comment]
@ilbiondo What is the status of this PR?

**Comment by alalazo** [comment]
Hi @ilbiondo I'm going to close the PR but feel free to reopen it if you want to continue working on that.

**Pull request closed without merging**